### ISSUE TITLE
[gateway_swagger_compile] unable to find dependencies

### ISSUE THREAD

**Issue by bweston92** [stackb/rules_proto#159]
All other protobuf compilers are working but for some reason the swagger compiler doesn't.

`BUILD` file contents.

```
proto_library(
    name = "introducers_proto",
    srcs = ["introducers.proto"],
    visibility = ["//visibility:public"],
    deps = [
        "//api/internalwkt:internalwkt_proto",
        "@com_google_protobuf//:empty_proto",
        "@com_google_protobuf//:timestamp_proto",
        "@go_googleapis//google/api:annotations_proto",
        "@com_github_grpc_ecosystem_grpc_gateway//protoc-gen-swagger/options:options_proto",
    ],
)

gateway_swagger_compile(
    name = "api_swagger",
    deps = [":introducers_proto"],
    verbose=4
)

```

Logs, as you can see it mentions them just seems to be blank though.

```
$ bazel build //api/introducers/v1:api_swagger
DEBUG: /home/bweston/.cache/bazel/_bazel_bweston/ef768c7103e8cc29b5e76fb15d2223c4/external/build_stack_rules_proto/compile.bzl:502:22: transitive source: <source file api/internalwkt/wkt.proto>
DEBUG: /home/bweston/.cache/bazel/_bazel_bweston/ef768c7103e8cc29b5e76fb15d2223c4/external/build_stack_rules_proto/compile.bzl:502:22: transitive source: <generated file external/com_google_protobuf/_virtual_imports/empty_proto/google/protobuf/empty.proto>
DEBUG: /home/bweston/.cache/bazel/_bazel_bweston/ef768c7103e8cc29b5e76fb15d2223c4/external/build_stack_rules_proto/compile.bzl:502:22: transitive source: <generated file external/com_google_protobuf/_virtual_imports/timestamp_proto/google/protobuf/timestamp.proto>
DEBUG: /home/bweston/.cache/bazel/_bazel_bweston/ef768c7103e8cc29b5e76fb15d2223c4/external/build_stack_rules_proto/compile.bzl:502:22: transitive source: <source file external/go_googleapis/google/api/annotations.proto>
DEBUG: /home/bweston/.cache/bazel/_bazel_bweston/ef768c7103e8cc29b5e76fb15d2223c4/external/build_stack_rules_proto/compile.bzl:502:22: transitive source: <source file external/go_googleapis/google/api/client.proto>
DEBUG: /home/bweston/.cache/bazel/_bazel_bweston/ef768c7103e8cc29b5e76fb15d2223c4/external/build_stack_rules_proto/compile.bzl:502:22: transitive source: <source file external/go_googleapis/google/api/field_behavior.proto>
DEBUG: /home/bweston/.cache/bazel/_bazel_bweston/ef768c7103e8cc29b5e76fb15d2223c4/external/build_stack_rules_proto/compile.bzl:502:22: transitive source: <source file external/go_googleapis/google/api/http.proto>
DEBUG: /home/bweston/.cache/bazel/_bazel_bweston/ef768c7103e8cc29b5e76fb15d2223c4/external/build_stack_rules_proto/compile.bzl:502:22: transitive source: <source file external/go_googleapis/google/api/resource.proto>
DEBUG: /home/bweston/.cache/bazel/_bazel_bweston/ef768c7103e8cc29b5e76fb15d2223c4/external/build_stack_rules_proto/compile.bzl:502:22: transitive source: <generated file external/com_google_protobuf/_virtual_imports/descriptor_proto/google/protobuf/descriptor.proto>
DEBUG: /home/bweston/.cache/bazel/_bazel_bweston/ef768c7103e8cc29b5e76fb15d2223c4/external/build_stack_rules_proto/compile.bzl:502:22: transitive source: <source file external/com_github_grpc_ecosystem_grpc_gateway/protoc-gen-swagger/options/annotations.proto>
DEBUG: /home/bweston/.cache/bazel/_bazel_bweston/ef768c7103e8cc29b5e76fb15d2223c4/external/build_stack_rules_proto/compile.bzl:502:22: transitive source: <source file external/com_github_grpc_ecosystem_grpc_gateway/protoc-gen-swagger/options/openapiv2.proto>
DEBUG: /home/bweston/.cache/bazel/_bazel_bweston/ef768c7103e8cc29b5e76fb15d2223c4/external/build_stack_rules_proto/compile.bzl:502:22: transitive source: <generated file external/com_google_protobuf/_virtual_imports/any_proto/google/protobuf/any.proto>
DEBUG: /home/bweston/.cache/bazel/_bazel_bweston/ef768c7103e8cc29b5e76fb15d2223c4/external/build_stack_rules_proto/compile.bzl:502:22: transitive source: <generated file external/com_google_protobuf/_virtual_imports/struct_proto/google/protobuf/struct.proto>
DEBUG: /home/bweston/.cache/bazel/_bazel_bweston/ef768c7103e8cc29b5e76fb15d2223c4/external/build_stack_rules_proto/compile.bzl:559:14: ProtoCompile: bazel-out/host/bin/external/com_google_protobuf/protoc --descriptor_set_out=bazel-out/k8-fastbuild/bin/api/introducers/v1/api_swagger/descriptor.source.bin --proto_path=bazel-out/k8-fastbuild/bin/api/introducers/v1/api_swagger --include_imports --include_source_info --swagger_out=bazel-out/k8-fastbuild/bin/api/introducers/v1/api_swagger --plugin=protoc-gen-swagger=bazel-out/host/bin/external/grpc_ecosystem_grpc_gateway/protoc-gen-swagger/protoc-gen-swagger_/protoc-gen-swagger bazel-out/k8-fastbuild/bin/api/introducers/v1/api_swagger/api/introducers/v1/introducers.proto
DEBUG: /home/bweston/.cache/bazel/_bazel_bweston/ef768c7103e8cc29b5e76fb15d2223c4/external/build_stack_rules_proto/compile.bzl:567:18: expected output:  bazel-out/k8-fastbuild/bin/api/introducers/v1/api_swagger/api/introducers/v1/introducers.swagger.json
INFO: Analyzed target //api/introducers/v1:api_swagger (0 packages loaded, 0 targets configured).
INFO: Found 1 target...
ERROR: /home/bweston/workspace/src/lwebco.de/cosmic/api/introducers/v1/BUILD.bazel:24:24: error executing shell command: '/bin/bash -c env && echo '
##### SANDBOX BEFORE RUNNING PROTOC' && find . && bazel-out/host/bin/external/com_google_protobuf/protoc --descriptor_set_out=bazel-out/k8-fastbuild/bin/api/introducers/v...' failed (Exit 1) bash failed: error executing command /bin/bash -c ... (remaining 1 argument(s) skipped)

Use --sandbox_debug to see verbose messages from the sandbox
google/protobuf/descriptor.proto: File not found.
google/api/annotations.proto:20:1: Import "google/protobuf/descriptor.proto" was not found or had errors.
google/api/annotations.proto:28:8: "google.protobuf.MethodOptions" is not defined.
google/protobuf/empty.proto: File not found.
google/protobuf/timestamp.proto: File not found.
google/protobuf/any.proto: File not found.
google/protobuf/struct.proto: File not found.
protoc-gen-swagger/options/openapiv2.proto:7:1: Import "google/protobuf/any.proto" was not found or had errors.
protoc-gen-swagger/options/openapiv2.proto:8:1: Import "google/protobuf/struct.proto" was not found or had errors.
protoc-gen-swagger/options/openapiv2.proto: "google.protobuf.Value" is not defined.
protoc-gen-swagger/options/openapiv2.proto: "google.protobuf.Value" is not defined.
protoc-gen-swagger/options/openapiv2.proto: "google.protobuf.Value" is not defined.
protoc-gen-swagger/options/openapiv2.proto: "google.protobuf.Value" is not defined.
protoc-gen-swagger/options/openapiv2.proto:153:3: "google.protobuf.Any" is not defined.
protoc-gen-swagger/options/openapiv2.proto: "google.protobuf.Value" is not defined.
protoc-gen-swagger/options/annotations.proto:7:1: Import "google/protobuf/descriptor.proto" was not found or had errors.
protoc-gen-swagger/options/annotations.proto:8:1: Import "protoc-gen-swagger/options/openapiv2.proto" was not found or had errors.
protoc-gen-swagger/options/annotations.proto:10:8: "google.protobuf.FileOptions" is not defined.
protoc-gen-swagger/options/annotations.proto:17:8: "google.protobuf.MethodOptions" is not defined.
protoc-gen-swagger/options/annotations.proto:24:8: "google.protobuf.MessageOptions" is not defined.
protoc-gen-swagger/options/annotations.proto:31:8: "google.protobuf.ServiceOptions" is not defined.
protoc-gen-swagger/options/annotations.proto:38:8: "google.protobuf.FieldOptions" is not defined.
api/introducers/v1/introducers.proto:6:1: Import "google/api/annotations.proto" was not found or had errors.
api/introducers/v1/introducers.proto:7:1: Import "google/protobuf/empty.proto" was not found or had errors.
api/introducers/v1/introducers.proto:8:1: Import "google/protobuf/timestamp.proto" was not found or had errors.
api/introducers/v1/introducers.proto:9:1: Import "protoc-gen-swagger/options/annotations.proto" was not found or had errors.
api/introducers/v1/introducers.proto:145:3: "google.protobuf.Timestamp" is not defined.
api/introducers/v1/introducers.proto:121:68: "google.protobuf.Empty" is not defined.
PWD=/home/bweston/.cache/bazel/_bazel_bweston/ef768c7103e8cc29b5e76fb15d2223c4/sandbox/linux-sandbox/223/execroot/de_lwebcode_cosmic
TMPDIR=/tmp
SHLVL=1
_=/usr/bin/env

##### SANDBOX BEFORE RUNNING PROTOC
.
./bazel-out
./bazel-out/k8-fastbuild
./bazel-out/k8-fastbuild/bin
./bazel-out/k8-fastbuild/bin/api
./bazel-out/k8-fastbuild/bin/api/introducers
./bazel-out/k8-fastbuild/bin/api/introducers/v1
./bazel-out/k8-fastbuild/bin/api/introducers/v1/api_swagger
./bazel-out/k8-fastbuild/bin/api/introducers/v1/api_swagger/_virtual_imports
./bazel-out/k8-fastbuild/bin/api/introducers/v1/api_swagger/_virtual_imports/any_proto
./bazel-out/k8-fastbuild/bin/api/introducers/v1/api_swagger/_virtual_imports/any_proto/google
./bazel-out/k8-fastbuild/bin/api/introducers/v1/api_swagger/_virtual_imports/any_proto/google/protobuf
./bazel-out/k8-fastbuild/bin/api/introducers/v1/api_swagger/_virtual_imports/any_proto/google/protobuf/any.proto
./bazel-out/k8-fastbuild/bin/api/introducers/v1/api_swagger/_virtual_imports/timestamp_proto
./bazel-out/k8-fastbuild/bin/api/introducers/v1/api_swagger/_virtual_imports/timestamp_proto/google
./bazel-out/k8-fastbuild/bin/api/introducers/v1/api_swagger/_virtual_imports/timestamp_proto/google/protobuf
./bazel-out/k8-fastbuild/bin/api/introducers/v1/api_swagger/_virtual_imports/timestamp_proto/google/protobuf/timestamp.proto
./bazel-out/k8-fastbuild/bin/api/introducers/v1/api_swagger/_virtual_imports/struct_proto
./bazel-out/k8-fastbuild/bin/api/introducers/v1/api_swagger/_virtual_imports/struct_proto/google
./bazel-out/k8-fastbuild/bin/api/introducers/v1/api_swagger/_virtual_imports/struct_proto/google/protobuf
./bazel-out/k8-fastbuild/bin/api/introducers/v1/api_swagger/_virtual_imports/struct_proto/google/protobuf/struct.proto
./bazel-out/k8-fastbuild/bin/api/introducers/v1/api_swagger/_virtual_imports/descriptor_proto
./bazel-out/k8-fastbuild/bin/api/introducers/v1/api_swagger/_virtual_imports/descriptor_proto/google
./bazel-out/k8-fastbuild/bin/api/introducers/v1/api_swagger/_virtual_imports/descriptor_proto/google/protobuf
./bazel-out/k8-fastbuild/bin/api/introducers/v1/api_swagger/_virtual_imports/descriptor_proto/google/protobuf/descriptor.proto
./bazel-out/k8-fastbuild/bin/api/introducers/v1/api_swagger/_virtual_imports/empty_proto
./bazel-out/k8-fastbuild/bin/api/introducers/v1/api_swagger/_virtual_imports/empty_proto/google
./bazel-out/k8-fastbuild/bin/api/introducers/v1/api_swagger/_virtual_imports/empty_proto/google/protobuf
./bazel-out/k8-fastbuild/bin/api/introducers/v1/api_swagger/_virtual_imports/empty_proto/google/protobuf/empty.proto
./bazel-out/k8-fastbuild/bin/api/introducers/v1/api_swagger/api
./bazel-out/k8-fastbuild/bin/api/introducers/v1/api_swagger/api/introducers
./bazel-out/k8-fastbuild/bin/api/introducers/v1/api_swagger/api/introducers/v1
./bazel-out/k8-fastbuild/bin/api/introducers/v1/api_swagger/api/introducers/v1/introducers.proto
./bazel-out/k8-fastbuild/bin/api/introducers/v1/api_swagger/api/internalwkt
./bazel-out/k8-fastbuild/bin/api/introducers/v1/api_swagger/api/internalwkt/wkt.proto
./bazel-out/k8-fastbuild/bin/api/introducers/v1/api_swagger/google
./bazel-out/k8-fastbuild/bin/api/introducers/v1/api_swagger/google/api
./bazel-out/k8-fastbuild/bin/api/introducers/v1/api_swagger/google/api/http.proto
./bazel-out/k8-fastbuild/bin/api/introducers/v1/api_swagger/google/api/field_behavior.proto
./bazel-out/k8-fastbuild/bin/api/introducers/v1/api_swagger/google/api/resource.proto
./bazel-out/k8-fastbuild/bin/api/introducers/v1/api_swagger/google/api/client.proto
./bazel-out/k8-fastbuild/bin/api/introducers/v1/api_swagger/google/api/annotations.proto
./bazel-out/k8-fastbuild/bin/api/introducers/v1/api_swagger/protoc-gen-swagger
./bazel-out/k8-fastbuild/bin/api/introducers/v1/api_swagger/protoc-gen-swagger/options
./bazel-out/k8-fastbuild/bin/api/introducers/v1/api_swagger/protoc-gen-swagger/options/openapiv2.proto
./bazel-out/k8-fastbuild/bin/api/introducers/v1/api_swagger/protoc-gen-swagger/options/annotations.proto
./bazel-out/host
./bazel-out/host/bin
./bazel-out/host/bin/external
./bazel-out/host/bin/external/com_google_protobuf
./bazel-out/host/bin/external/com_google_protobuf/protoc
./bazel-out/host/bin/external/com_google_protobuf/protoc.runfiles
./bazel-out/host/bin/external/com_google_protobuf/protoc.runfiles/com_google_protobuf
./bazel-out/host/bin/external/com_google_protobuf/protoc.runfiles/com_google_protobuf/protoc
./bazel-out/host/bin/external/com_google_protobuf/protoc.runfiles/de_lwebcode_cosmic
./bazel-out/host/bin/external/com_google_protobuf/protoc.runfiles/de_lwebcode_cosmic/external
./bazel-out/host/bin/external/com_google_protobuf/protoc.runfiles/de_lwebcode_cosmic/external/com_google_protobuf
./bazel-out/host/bin/external/com_google_protobuf/protoc.runfiles/de_lwebcode_cosmic/external/com_google_protobuf/protoc
./bazel-out/host/bin/external/grpc_ecosystem_grpc_gateway
./bazel-out/host/bin/external/grpc_ecosystem_grpc_gateway/protoc-gen-swagger
./bazel-out/host/bin/external/grpc_ecosystem_grpc_gateway/protoc-gen-swagger/protoc-gen-swagger_
./bazel-out/host/bin/external/grpc_ecosystem_grpc_gateway/protoc-gen-swagger/protoc-gen-swagger_/protoc-gen-swagger
Target //api/introducers/v1:api_swagger failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 0.172s, Critical Path: 0.05s
INFO: 0 processes.
FAILED: Build did NOT complete successfully

```

**Comment by bweston92** [comment]
Seems like there is a virtual imports folder which isn't read?

```
~/.cache/bazel/.../bazel-out/k8-fastbuild/bin/api/gateways/customers/v1/api_swagger$ find . | grep proto
./_virtual_imports/any_proto
./_virtual_imports/any_proto/google
./_virtual_imports/any_proto/google/protobuf
./_virtual_imports/any_proto/google/protobuf/any.proto
./_virtual_imports/timestamp_proto
./_virtual_imports/timestamp_proto/google
./_virtual_imports/timestamp_proto/google/protobuf
./_virtual_imports/timestamp_proto/google/protobuf/timestamp.proto
./_virtual_imports/struct_proto
./_virtual_imports/struct_proto/google
./_virtual_imports/struct_proto/google/protobuf
./_virtual_imports/struct_proto/google/protobuf/struct.proto
./_virtual_imports/descriptor_proto
./_virtual_imports/descriptor_proto/google
./_virtual_imports/descriptor_proto/google/protobuf
./_virtual_imports/descriptor_proto/google/protobuf/descriptor.proto
./_virtual_imports/empty_proto
./_virtual_imports/empty_proto/google
./_virtual_imports/empty_proto/google/protobuf
./_virtual_imports/empty_proto/google/protobuf/empty.proto
./api/gateways/customers/v1/customers.proto
./api/internalwkt/wkt.proto
./google/api/http.proto
./google/api/field_behavior.proto
./google/api/resource.proto
./google/api/client.proto
./google/api/annotations.proto
./protoc-gen-swagger
./protoc-gen-swagger/options
./protoc-gen-swagger/options/openapiv2.proto
./protoc-gen-swagger/options/annotations.proto
```

**Comment by bweston92** [comment]
fixed in #152 I hadn't updated in a while
